### ISSUE TITLE
Microsoft.Extensions.AI has no documentation nor Sample for Tool Usage

### ISSUE THREAD

**Issue by JohnGalt1717** [dotnet/extensions#5739]
### Description

IChatClient clearly supports tool usage, but there's no documentation that I can find anywhere (I can't find real documentation for Microsoft.Extensions.AI, only the announcement) on this topic, and Copilot and Github Copilot are also clueless on how to use tools properly.

### Reproduction Steps

Try and use tools with Microsoft.Extensions.AI.

### Expected behavior

There should be a proper documentation page, the announcement should reference it, the nuget package should have that as the package home, and the documentation section should have explicit instructions on how to use tools.

### Actual behavior

None of that is available.

### Known Workarounds

🤷

### Configuration

.net 9,

<PackageVersion Include="Microsoft.Extensions.AI" Version="9.0.1-preview.1.24570.5" />

### Other information

...

**Comment by stephentoub** [comment]
No doubt samples and docs can be improved, but you can see tool use examples in the package readme:
https://www.nuget.org/packages/Microsoft.Extensions.AI.Abstractions

in the eshop sample:
https://github.com/dotnet/eShop/blob/main/src/WebApp/Components/Chatbot/ChatState.cs#L41-L47

in the dotnet/Ai-samples, eg
https://github.com/dotnet/ai-samples/blob/main/src/microsoft-extensions-ai/ollama/OllamaExamples/ToolCalling.cs

in the eShopSupport sample, eg
https://github.com/dotnet/eShopSupport/blob/690d0815444539afb120fb9145253e990fbcfb0d/src/Backend/Api/AssistantApi.cs#L56

In short, set the Chat Options.Tools to the collection of tools you want advertised to the llm. And if you want them automatically invoked by the client when they're requested, add that support to the IChatClient, typically via UseFunctionInvocation.

**Comment by stephentoub** [comment]
cc: @LadyNaggaga

**Comment by JohnGalt1717** [comment]
The problem is that this is so undiscoverable that even with Copilot doing web searches, it still can't find it, and if it can't find it, no one else will either.

It also doesn't talk about Async, or any instructions on how to get good results from the tools (like meta data for parameters, DescriptionAttribute, JsonSchema) so one is left to stumble around in the dark.

All of the documentation should be in the .net documentation like everything else too.

**Comment by jeffhandley** [comment]
Thanks for the feedback on this, @JohnGalt1717. We're still in preview milestones and we will have documentation integrated as you recommended when we approach a stable release. Thanks for the details on what you expected to find.

**Comment by JohnGalt1717** [comment]
Do you want to keep this open to track or should I close?

**Comment by jeffhandley** [comment]
We'll leave it open for now until we have a broader docs issue in place that incorporates your specific scenario. Thanks!

**Comment by jeffhandley** [comment]
@JohnGalt1717 -- We now have this documentation available: [Microsoft.Extensions.AI libraries (Preview) - .NET | Microsoft Learn](https://learn.microsoft.com/dotnet/ai/microsoft-extensions-ai#tool-calling)
